src/Kafka/Producer/Callbacks.hs

```diff
@@ -45,10 +45,12 @@ deliveryCallback callback kc = rdKafkaConfSetDrMsgCb (getRdKafkaConf kc) realCb
       if cbPtr == nullPtr then
         pure ()
       else do
-        msgCb <- deRefStablePtr @(DeliveryReport -> IO ()) $ castPtrToStablePtr $ cbPtr
+        let stablePtr = castPtrToStablePtr cbPtr
+        msgCb <- deRefStablePtr @(DeliveryReport -> IO ()) stablePtr
         -- Here we fork the callback since it might be a longer action and
         -- blocking here would block librdkafka from continuing its execution
         void . forkIO $ msgCb rep
+        freeStablePtr stablePtr
```

Comment:
Would `bracket` be better in this situation? 
Or it doesn't matter in this case?

Reply:
It would be nice, but I don't think it's possible. The pointer is created in `Kafka.Producer` and when the per-message callback is invoked it ends up here. So there's not really a good way to tie those two together that I can think of...

Reply:
I thought of `bracket (castPtrToStablePtr cbPtr) freeStablePtr $ \stablePrt -> do ...`, not the actual pointer...

Reply:
We'd have to do:

```haskell
      else bracket (pure $ castPtrToStablePtr cbPtr) freeStablePtr $ \stablePtr -> do
        msgCb <- deRefStablePtr @(DeliveryReport -> IO ()) stablePtr
        -- Here we fork the callback since it might be a longer action and
        -- blocking here would block librdkafka from continuing its execution
        void . forkIO $ msgCb rep
```
I can change it to that if ya like :)

Reply:
Done